doc/source/release/1.16.0-notes.rst

```diff
@@ -17,114 +17,1035 @@ run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
-  can help it for keep going and find a better solution.
+  trust region radii given by ``rhobeg`` and ``tol`` respectively. A larger
+  ``rhobeg`` can help the algorithm take bigger steps initially, while a
+  smaller ``tol`` can help it continue and find a better solution.
   For more information, see the `PRIMA documentation <https://www.libprima.net>`_.
+- Several of the ``minimize`` methods, and the ``least_squares`` function,
+  have been given a ``workers`` keyword. This allows parallelization of some
+  calculations via a map-like callable, such as ``multiprocessing.Pool``. These
+  parallelization opportunities typically occur during numerical
+  differentiation. This can greatly speed-up minimization when the objective
+  function is expensive to calculate.
+- The ``lm`` method of ``least_squares`` can now accept ``3-point`` and ``cs``
+  for the ``jac`` keyword.
+- SLSQP Fortran77 code ported to C. When this method is used now the constraint
+  multipliers are exposed to the user through the ``multiplier`` keyword of
+  the returned ``OptimizeResult`` object.
+- NNLS code has been corrected and rewritten in C to address the performance
+  regression introduced in 1.15.x
+- ``optimize.root`` now warns for invalid inner parameters when using the
+  ``newton_krylov`` method
+- The return value of minimization with ``method='L-BFGS-B'`` now has
+  a faster ``hess_inv.todense()`` implementation. Time complexity has improved
+  from cubic to quadratic.
+- ``optimize.least_squares`` has a new ``callback`` argument that is applicable
+  to ``trf`` and ``dogbox`` methods. ``callback`` may be used to track
+  optimization results at each step or to provide custom conditions for
+  stopping.
 
 
 ``scipy.signal`` improvements
 =============================
+- A new function ``firwin_2d`` for the creation of a 2-D FIR Filter using the
+  1-D window method was added.
+- ``signal.cspline1d_eval`` and ``signal.qspline1d_eval`` now provide an
+  informative error on empty input rather than hitting the recursion limit.
+- A new function `scipy.signal.closest_STFT_dual_window` to calculate the
+  ``ShortTimeFFT`` dual window of a given window closest to a desired dual
+  window.
+- A new classmethod `scipy.signal.ShortTimeFFT.from_win_equals_dual` to
+  create a ``ShortTimeFFT`` instance where the window and its dual are equal
+  up to a scaling factor. It allows to create short-time Fourier transforms
+  which are unitary mappings.
+- The performance of ``signal.convolve2d`` has improved.
 
 
 ``scipy.sparse`` improvements
 =============================
+- ``coo_array`` supports nD arrays using binary and reduction operations.
+- Math is faster between two DIA arrays/matrices: add, sub, multiply, matmul.
+- ``csgraph.dijkstra`` shortest_path is more efficient.
+- ``csgraph.yen`` has performance improvements.
+- Import of ``_lib.issparse`` allows checking for ``scipy.sparse`` object
+  without full import of ``scipy``.
```

Comment:
This should be removed or rewritten I think. The private utility function isn't relevant to readers of release notes, so we should express it as the change in behavior for public functions (if relevant).